File: 코딩 기초 트레이닝/조건에 맞게 수열 반환하기2-p181881/조건에 맞게 수열 반환하기2.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181881
function solution(arr) {
    let answer = 0;
    let prev = [];
    let current = arr;

    const transform = (arr) => {
        arr = arr.map((a) => {
            if(a >= 50 && a % 2 === 0) {
                return a / 2;
            } else if (a <= 50 && a % 2 !== 0) {
                return a * 2 + 1;
            } else {
                return a;
            }
        });

        return arr;
    }

    while(prev.join('') !== current.join('')) {
        prev = current;
        current = transform(prev);
        answer += 1;
    }


    return answer -1;
}

File: 코딩 기초 트레이닝/조건에 맞게 수열 반환하기2-p181881/조건에 맞게 수열 반환하기2.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181881

def solution(arr):
    # a >= 50 짝수 /2
    # a >= 50 홀수 *2 +1
    answer = 0

    flag = True

    while flag:
        inner_flag = False
        for i, a in enumerate(arr):
            if a >= 50 and a % 2 == 0:
                arr[i] = a/2
                inner_flag = True
            elif a <= 50 and a % 2 != 0:
                arr[i] = a*2 +1
                inner_flag = True

        if inner_flag :
            answer += 1
        else:
            flag = False

    return answer
